core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -84,11 +82,13 @@ public void addDayActivity(DayActivity dayActivity)
 			throw new IllegalStateException("Week is already full (" + dayActivities.size() + " days present)");
 		}
 
+		dayActivity.setWeekActivity(this);
 		dayActivities.add(dayActivity);
 	}
 
 	public void removeAllDayActivities()
 	{
+		dayActivities.forEach(da -> da.setWeekActivity(null));
```

Comment:
I guess this has to do with the orphan removal. Why do you not just call delete on the `DayActivityRepository`?